build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -4,294 +4,187 @@
  */
 
 import * as fs from "node:fs/promises";
-import { isInternalVersionRange } from "@fluid-tools/version-tools";
-import { Logger } from "@fluidframework/build-tools";
+import path from "node:path";
+import { isInternalVersionRange, isInternalVersionScheme } from "@fluid-tools/version-tools";
+import type { Logger } from "@fluidframework/build-tools";
 import { Flags } from "@oclif/core";
-import fetch from "node-fetch";
 import { BaseCommand } from "../../base";
-import { PackageVersionList } from "../../library";
-
-// Define the interface for build details
-interface IBuildDetails {
-	definition: { name: string };
-	status: string;
-	result: string;
-	sourceBranch: string;
-	finishTime: string;
-	buildNumber: string;
-}
+import type { PackageVersionList } from "../../library";
 
 export class UnreleasedReportCommand extends BaseCommand<typeof UnreleasedReportCommand> {
 	static readonly description =
-		`Creates a release report for the most recent build of the client release group published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report is a combination of the "simple" and "caret" report formats. Packages released as part of the client release group will have an exact version range, while other packages, such as server packages or independent packages, will have a caret-equivalent version range.`;
+		`This command creates a release report for each build of the client release group published to an internal ADO feed. It utilizes the version set in the pipeline run to generate reports in the "simple" and "caret" formats. This command is used within the "upload-dev-manifest" yaml file and currently operates exclusively for non-PR main branch builds.`;
 
 	static readonly flags = {
-		repo: Flags.string({
-			description: "Repository name",
-			required: true,
-		}),
-		ado_pat: Flags.string({
+		version: Flags.string({
 			description:
-				"ADO Personal Access Token. This flag should be provided via the ADO_PAT environment variable for security reasons.",
+				"Version to generate a report for. Typically, this version is the version of a dev build.",
 			required: true,
-			env: "ADO_PAT",
+			char: "V",
 		}),
-		sourceBranch: Flags.string({
-			description: "Branch name across which the dev release manifest should be generated.",
+		outDir: Flags.directory({
+			description: "Release report output directory",
 			required: true,
+			char: "o",
 		}),
-		output: Flags.string({
-			description: "Output manifest file path",
-			required: true,
+		caretManifestFilePath: Flags.string({
+			description: "Specify the path to the caret manifest file",
+			char: "c",
+			default: ".",
+		}),
```

Comment:
```suggestion
		caretManifestFilePath: Flags.file({
			description: "Path to a report file in the 'caret' format.",
			exists: true,
		}),
```

Use the file flag and make oclif check the file exists for you.

You may need to update the release report command so that you can output specific file names. Basically avoid searching for the files you need - rather make it so the file paths are always provided - which _might_ require changing release report to optionally output names without variables like the version.

Maybe release report could take a --baseFileName flag and if provided it outputs the files with names like `baseFileName.caret.json`?

Reply:
That's a good idea. I have pushed the changes, but code cleanup is remaining. I'll work on that in sometime.